src/cpp/rtps/reader/StatefulReader.cpp

```diff
@@ -550,8 +550,10 @@ bool StatefulReader::change_removed_by_history(
         {
             if(!a_change->isRead && wp->available_changes_max() >= a_change->sequenceNumber)
             {
-                assert(total_unread_ > 0);
-                --total_unread_;
```

Comment:
What was the deal here? Was it that `total_unread_` is unsigned or something like that?

Reply:
Unsigned or not, the system can be unstable if I decrease this variable when its value is zero. And it can occur the way I've implemented it.

Reply:
:+1: